module/zfs/vdev_rebuild.c

```diff
@@ -307,6 +309,15 @@ vdev_rebuild_complete_sync(void *arg, dmu_tx_t *tx)
 	vdev_rebuild_phys_t *vrp = &vr->vr_rebuild_phys;
 
 	mutex_enter(&vd->vdev_rebuild_lock);
+
+	/* Handle the lost reset wanted signal */
+	if (vd->vdev_rebuild_reset_wanted) {
+		dsl_sync_task_nowait(spa_get_dsl(spa), vdev_rebuild_reset_sync,
+		    (void *)(uintptr_t)vd->vdev_id, tx);
```

Comment:
Changed to 
```C
if (vd->vdev_rebuild_reset_wanted) {
		mutex_exit(&vd->vdev_rebuild_lock);
		vdev_rebuild_reset_sync((void *)(uintptr_t)vd->vdev_id, tx);
		return;
	}
```
I think, the lock is still necessary for the `vdev_rebuild_reset_wanted` as it can get changed by the 2nd `vdev_rebuild` before it reaches the `vdev_rebuild_lock` in `vdev_rebuild_complete_sync`.